### PR TITLE
Even more Application API

### PR DESCRIPTION
**Pull request by jgustie**
This PR addresses a few gaps in the Application API:
* The endpoint URL is not configurable, this is necessary to "re-home" the existing Experiments API on top of the Application API. As fallout from this, I made use of more specific URL functions rather then relying on string manipulation. This also address an edge case for testing where you can inject query parameters to the base URL.
* Added some missing relations.
* Added a helper to test for activity item tags to help with some awkwardness of using standard JSON feed.

### CODE REVIEW

File: pkg/api/client.go [hunk 1]
```diff
@@ -57,13 +55,13 @@ type httpClient struct {
...
-
-	p := c.endpoint.Path + "/" + strings.TrimLeft(ep, "/")
-	u := *c.endpoint
-	u.Path = p
-	return &u
+	u, err := c.endpoint.Parse(ep)
+	if err != nil {
+		// If code panics here, the caller needs to verify it's input before
+		// passing it on to the `Client.URL(endpoint string)` function.
+		panic(err)
```
Comment (bradbeam) [inline pkg/api/client.go:62]: is panic used here instead of an error to not have to refactor all callers to handle an error? 
(always suspect of panics)

**Review by bradbeam — APPROVED** [review]

File: pkg/api/client.go (hunk 1 above)
Comment (jgustie) [reply pkg/api/client.go:62]: Yes-ish. It would make the callers all really obnoxious because they usually just take `URL(...).String()` as a one-liner; but also the endpoint used to be a constant so we always knew it was a valid URL. I think that is a good pre-condition to maintain.
